libpimeval/src/pimPerfEnergyModels.cpp

```diff
@@ -10,8 +10,40 @@
 #include "pimPerfEnergyTables.h"
 #include <cstdio>
 
-//! @brief  pimParamsPerf ctor
-pimParamsPerf::pimParamsPerf(pimParamsDram* paramsDram)
+
+//! @brief  A factory function to get perf energy model for sim target
+pimPerfEnergyBase*
+pimPerfEnergyFactory::getPerfEnergyModel(PimDeviceEnum simTarget, pimParamsDram* paramsDram)
+{
+  pimPerfEnergyBase* model = nullptr;
+  switch (simTarget) {
+    case PIM_DEVICE_BITSIMD_V:
+    case PIM_DEVICE_BITSIMD_V_AP:
+    case PIM_DEVICE_BITSIMD_H:
+    case PIM_DEVICE_SIMDRAM:
+      std::printf("PIM-Info: Created performance energy model for bit-serial PIM\n");
+      model = new pimPerfEnergyBitSerial(paramsDram);
```

Comment:
I would prefer smart pointers instead of `new` and `delete` to avoid manual memory management.